### PR TITLE
1025/caching cloud

### LINKED ISSUE #1025: Migrate reading AWS and GCP data catalogs to V1
### Kind of request

None

### Enhancement Description

There are still some skipped tests in `test_data_adapter.py`. These need to be migrated to v1. This is not trivial as the `caching` module does not work well with this data.

### Use case

Reading these catalogs in v1.

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Jaapel**
## Issue addressed
Fixes #1025 

## Explanation
Wanted to fix the AWS data catalog.
Noticed that the dataset tested used a gdal virtual filesystem. This messes with the caching settings. Of course this path is different in windows and linux, so the code had to be revised.
Then that the .vrt caching did not cache at all: the `intersects` in-line functions was not correct and skipped all caching.
The vrt also had to be altered to account for the cached files.

## General Checklist
- [x] Updated tests or added new tests
- [x] Branch is up to date with `v1`
- [x] Tests & pre-commit hooks pass
- [x] Updated documentation
- [x] Updated changelog.rst

## Additional Notes (optional)
I did change the data catalog for GCP CMIP6 data. But as we are still in alpha, i do not release a new version.

### CODE REVIEW

File: tests/data_catalog/test_data_catalog.py [hunk 1]
```diff
@@ -821,6 +824,69 @@ def test_s3(self, data_catalog: DataCatalog):
...
+        )
+        # Check reading and some preprocess
+        assert "precip" in ds
+        assert not np.any(ds[ds.raster.x_dim] > 180)
+        # Skip as I don't think this adds value to testing a gcs cloud archive
+        # Write and compare
+        # fn_nc = str(tmpdir.join("test.nc"))
+        # ds.to_netcdf(fn_nc)
+        # ds1 = data_catalog.get_rasterdataset(fn_nc)
+        # assert np.allclose(ds["precip"][0, :, :], ds1["precip"][0, :, :])
```
Comment (savente93) [inline tests/data_catalog/test_data_catalog.py:888]: I think we can remove this right?

**Review by savente93 — COMMENTED** [review]
Looking good Jaap. I don't know enough about how vrt files work to comment very semantically but the code looks good to me. I have a few small questions, once those are resolved I'll be happy to approve

File: tests/conftest.py [hunk 2]
```diff
@@ -49,6 +50,15 @@ def PLUGINS() -> Plugins:
...
 
+@pytest.fixture()
+def test_settings(tmp_dir: Path) -> Generator[Settings, None, None]:
+    """Temporary sets settings for testing."""
+    cache_dir: Path = tmp_dir / "test_caching"
+    SETTINGS.cache_root = cache_dir
+    yield SETTINGS
+    SETTINGS.cache_root = SETTINGS.model_fields["cache_root"].default
+
+
```
[the remark this answers is not in the document]
Comment (Jaapel) [reply tests/conftest.py]: Yes, changed to `tmp_path` built-in fixture

File: hydromt/_utils/caching.py [hunk 3]
```diff
@@ -63,54 +112,101 @@ def _cache_vrt_tiles(
...
-            if not isfile(dst):
-                paths.append((src, dst))
-    # TODO multi thread download
+    for source in band_name.findall(source_name):
+        if geom is None or _overlaps(source, transform, bbox):
+            vrt_ref_el: Optional[ET.Element] = source.find("SourceFilename")
+            if vrt_ref_el is None:
+                raise ValueError(f"Could not find Source File in vrt: {vrt_uri}.")
+            vrt_ref: str = vrt_ref_el.text.replace("\\", os.sep)  # dewindowsify
+            vrt_relative: int = int(vrt_ref_el.get("relativeToVRT"))  # 0 or 1
```
Comment (savente93) [inline hydromt/_utils/caching.py:166]: why int and not bool?

**Review by savente93 — APPROVED** [review]
LGTM

File: hydromt/_utils/caching.py (hunk 3 above)
Comment (Jaapel) [reply hydromt/_utils/caching.py:166]: The source format gives either 0 or 1, I like to represent it that way.

**Pull request merged**